### PR TITLE
Social-media-website-mybranch

### PR DESCRIPTION
**Pull request by tahsinrukaiya**
Second pull request

### CODE REVIEW

File: src/JS/auth/login.mjs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                    body: JSON.stringify(data),
+                };
+                const response = await fetch(url, postData);
+                console.log(response);
+                console.log("Log in Successfull!", response);
+
+                window.location.href = "../feed/index.html";
```
Comment (sulenchy) [inline src/JS/auth/login.mjs]: This should go in the else block

File: src/JS/auth/login.mjs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+                if (!response.ok) {
+                    throw new Error("Network Issue");
+                }
+                else {
+                    const json = await response.json();
+                    // After successful login
+                    saveStorage("token", json.token);
```
Comment (sulenchy) [inline src/JS/auth/login.mjs]: This should be 
```suggestion
                    saveStorage("token", json.AccessToken);
```

File: src/JS/auth/register.mjs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                    body: JSON.stringify(data),
+                };
+                const response = await fetch(url, postData);
+                console.log(response);
+                const json = await response.json();
+                console.log(json);
+                console.log("Registration Successfull!", response);
+                alert("Registration Successfull!");
+                window.location.href = "public/pages/logIn/index.html";
+                if (!response.ok) {
```
Comment (sulenchy) [inline src/JS/auth/register.mjs:30]: This should be after line 35. The validation should come before the usage

File: src/JS/feed/create_post.mjs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+const post_form = document.getElementById('post_form');
+const create_post_btn = document.getElementById('create_post_btn');
+const title = document.getElementById('title');
+const body = document.getElementById('body');
+const tags = document.getElementById('tags');
+const media = document.getElementById('media');
+
+
+// Retrieve the access token from local storage
+const storedAccessToken = localStorage.getItem('accessToken');
```
Comment (sulenchy) [inline src/JS/feed/create_post.mjs]: This should be

```suggestion
const storedAccessToken = localStorage.getItem('token');
```

**Review by sulenchy — CHANGES_REQUESTED** [review]
@tahsinrukaiya It is evident that you have put some effort into this delivery. Unfortunately, it did not pass. Most of the JavaScript functionalities are not working due to the use of incorrect variables and some typos.. In some areas, you redirect before the authentication.

**Comment by tahsinrukaiya** [comment]
Hi Abidun,
Thank you for your feedback. I saw your comments and I feel so sad that the assignment is not accepted after so many efforts. Could you please tell me what should i do to pass then?

Thank you


Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Pull request closed without merging**